app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -230,83 +218,85 @@ def parse_availability(availabilities, resources_by_metric_id)
         resources_by_metric_id
       end
 
-      def parse_deployment(server, deployment)
-        specific = {
-          :name              => parse_deployment_name(deployment.id),
-          :middleware_server => server,
-        }
-        parse_base_item(deployment).merge(specific)
+      def parse_deployment(deployment, inventory_object)
+        parse_base_item(deployment, inventory_object)
+        inventory_object.name = parse_deployment_name(deployment.id)
+        inventory_object
       end
 
-      def parse_messaging(server, messaging, config)
-        specific = {
-          :name              => messaging.name,
-          :middleware_server => server,
-          :messaging_type    => messaging.to_h['type']['name']
-        }
+      def parse_messaging(messaging, inventory_object, config)
+        parse_base_item(messaging, inventory_object)
+
+        inventory_object.name = messaging.name
+        inventory_object.messaging_type = messaging.to_h['type']['name']
+
         if !config.empty? && !config['value'].empty? && config['value'].respond_to?(:except)
-          specific[:properties] = config['value'].except('Username', 'Password')
+          inventory_object.properties = config['value'].except('Username', 'Password')
         end
-        parse_base_item(messaging).merge(specific)
+
+        inventory_object
```

Comment:
I see that this is not being used anywhere? So we don't need to return this object? Seems like this is present for all parse methods. Is there some reason for it?